pkg/config/instrumentation/remote_config_metrics.go

```diff
@@ -34,10 +40,12 @@ func newRemoteConfigMetrics() *remoteConfigMetrics {
 	return &remoteConfigMetrics
 }
 
-func (r *remoteConfigMetrics) InstrumentRemoteConfigFetch(code int) {
-	r.fetchStatusCodes.WithLabelValues(fmt.Sprintf("%d", code)).Inc()
+func InstrumentRemoteConfigFetch(code int) {
+	metricsInitializer.Do(initializeRemoteConfigMetrics)
```

Comment:
would it be easier to add `remoteConfigMetrics` to the `httpProvider` type and initialize it in `newHTTPProvider` (instead of globally)? or do we intent to use these metrics in other places?

Reply:
My thinking was that we may want to add some agent management-specific metrics here in the future, which would be used in `agentmanagement.go`